File: PAP2526/python/test_my_math.py
import unittest
import my_math


class TestMyMath(unittest.TestCase):
    
    def test_estPair(self):
        self.assertTrue(my_math.estPair(0))
        self.assertTrue(my_math.estPair(8))
        self.assertFalse(my_math.estPair(9))
        self.assertRaises(AssertionError, my_math.estPair, -9)
    
    def test_fact(self):
        self.assertEqual(1, my_math.fact(0))
        self.assertEqual(1, my_math.fact(1))
        self.assertEqual(120, my_math.fact(5))
        self.assertRaises(AssertionError, my_math.fact, -1)
    
    def test_nbZero(self): 
        self.assertEqual(1, my_math.nbZero(0))
        self.assertEqual(0, my_math.nbZero(98))
        self.assertEqual(2, my_math.nbZero(9080))
        self.assertRaises(AssertionError, my_math.nbZero, -1)
    
    def test_nbPassages(self):
        self.assertEqual(0, my_math.nbPassages(1))
        self.assertEqual(6, my_math.nbPassages(10))
        self.assertRaises(AssertionError, my_math.nbPassages, 0)
    
    def test_speedConverter(self):
        self.assertEqual(0.0, my_math.speedConverter(0, 'K', 'M'))
    
    def test_fibo(self):
        self.assertEqual(0, my_math.fibo(0))
        self.assertEqual(1, my_math.fibo(1))
        self.assertEqual(1, my_math.fibo(2))
        self.assertEqual(8, my_math.fibo(6))
        self.assertRaises(AssertionError, my_math.fibo, -1)
    
    def test_nombrerOr(self):
        self.assertEqual(1, my_math.nombreOr(1));
        self.assertEqual(2, my_math.nombreOr(2));
        self.assertAlmostEqual(1.666666, my_math.nombreOr(4), delta=0.000001);
        self.assertAlmostEqual(1.6, my_math.nombreOr(5), delta=0.000001);
        
    def test_estPremier(self):
        self.assertTrue(my_math.estPremier(2))
        self.assertTrue(my_math.estPremier(3))
        self.assertTrue(my_math.estPremier(31))
        self.assertFalse(my_math.estPremier(4))
        self.assertFalse(my_math.estPremier(25))
        self.assertFalse(my_math.estPremier(33))
        self.assertRaises(AssertionError, my_math.estPremier, 1)
        

if __name__ == "__main__":
    unittest.main()

File: PAP2526/test/tp01/TestMyMath.java
package tp01;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

import tp01.MyMath.SpeedUnit;

public class TestMyMath {

	@Test
	void testEstPair() {
		assertTrue(MyMath.estPair(0));
		assertTrue(MyMath.estPair(8));
		assertFalse(MyMath.estPair(9));
		assertThrows(AssertionError.class, () -> MyMath.estPair(-9));
	}

	@Test
	void testFact() {
		assertEquals(1, MyMath.fact(0));
		assertEquals(1, MyMath.fact(1));
		assertEquals(120, MyMath.fact(5));
		assertThrows(AssertionError.class, () -> MyMath.fact(-1));
		assertThrows(AssertionError.class, () -> MyMath.fact(21));
	}

	@Test
	void testNbZero() {
		assertEquals(1, MyMath.nbZero(0));
		assertEquals(0, MyMath.nbZero(123));
		assertEquals(3, MyMath.nbZero(10900));
		assertThrows(AssertionError.class, () -> MyMath.nbZero(-1));
	}

	@Test
	void testNbPassages() {
		assertEquals(6, MyMath.nbPassages(10));
		assertEquals(0, MyMath.nbPassages(1));
		assertThrows(AssertionError.class, () -> MyMath.nbPassages(0));
	}

	@Test
	void testSpeedConverter() {
		assertEquals(62.1371, MyMath.speedConverter(100.0, SpeedUnit.KMH, SpeedUnit.MILES), 0.001);
		assertEquals(53.9957, MyMath.speedConverter(100.0, SpeedUnit.KMH, SpeedUnit.NOEUD), 0.0001);
		assertEquals(185.2, MyMath.speedConverter(100.0, SpeedUnit.NOEUD, SpeedUnit.KMH), 0.0001);
		assertEquals(160.934, MyMath.speedConverter(100.0, SpeedUnit.MILES, SpeedUnit.KMH), 0.0001);
		assertEquals(86.8974, MyMath.speedConverter(100.0, SpeedUnit.MILES, SpeedUnit.NOEUD), 0.0001);
		assertEquals(115.07794, MyMath.speedConverter(100.0, SpeedUnit.NOEUD, SpeedUnit.MILES), 0.0001);

	}

	@Test
	void testFibo() {
		assertEquals(0, MyMath.fibo(0));
		assertEquals(1, MyMath.fibo(1));
		assertEquals(8, MyMath.fibo(6));
		assertThrows(AssertionError.class, () -> MyMath.fibo(-1));
	}

	@Test
	void testnbrOr() {
		assertEquals(1, MyMath.nombreOr(1));
		assertEquals(2, MyMath.nombreOr(2));
		assertEquals(1.666666, MyMath.nombreOr(4), 0.000001);
		assertEquals(1.6, MyMath.nombreOr(5), 0.000001);
		assertThrows(AssertionError.class, () -> MyMath.nombreOr(0));
	}

	@Test
	void testPremier() {
		assertTrue(MyMath.estPremier(2));
		assertTrue(MyMath.estPremier(3));
		assertTrue(MyMath.estPremier(5));
		assertTrue(MyMath.estPremier(7));
		assertTrue(MyMath.estPremier(31));
		assertFalse(MyMath.estPremier(4));
		assertFalse(MyMath.estPremier(9));
		assertFalse(MyMath.estPremier(25));
		assertFalse(MyMath.estPremier(33));

		assertThrows(AssertionError.class, () -> MyMath.estPremier(1));
	}
}
